src/algorithms/pathfinders/dijkstra.ts

```diff
@@ -0,0 +1,111 @@
+import EdgeCost from '../models/edgecost';
+import Graph from '../models/graph';
+import Vertex from '../models/vertex';
+/**
+ * Implements Dijkstra's algorithm to find the shortest path between 2 vertices
+ */
+export default class DijkstraPathfinder {
+    /**
+     * Stores best known distances from starting vertex to all the others
+     */
+    distances: Map<number, {
+        distance: number;
+        sourceID: number;
+    }>;
+
+    /**
+     * stores all unvisited vertices
+     */
+    unvisited: Set<number>;
+
+    /**
+     * Returns an array of vertex IDs that represents the shortest path between 2 vertices
+     * @param startID : ID of starting vertex
+     * @param endID : ID of destination vertex
+     * @param mygraph : relevant graph
+     */
+    FindPath(startID: number, endID: number, mygraph: Graph): number[] {
+        this.Initialize(mygraph);
+        this.distances.set(startID, {distance: 0, sourceID: -9001});

```

Comment:
Does this -9001 ID have some special purpose, meaning that there is no such ID? Why not use `null` then?